otter/convergence/service.py

```diff
@@ -531,6 +537,20 @@ def converge_one_group(currently_converging, recently_converged,
         # unexpected errors, so convergence will be retried.
         yield err(None, 'converge-non-fatal-error')
     else:
+        if result[0] is StepResult.LIMITED_RETRY:
+            # We allow further iterations to proceed as long as we haven't been
+            # waiting for a LIMITED_RETRY for 10 consecutive iterations.
+            if (yield waiting.read())[group_id] > 10:
```

Comment:
Can 10 be configurable value? Unfortunately, it might have to become another argument :expressionless: